app/controllers/sessions_controller.rb

```diff
@@ -1,34 +1,44 @@
 # frozen_string_literal: true
 
-# SessionsController
+# Sessions controller
 class SessionsController < ApplicationController
-  before_action :load_user, only: :create
+  before_action :load_user, :check_authenticate, :check_activation, only: :create
 
   def new; end
 
   def create
-    if @user.authenticate params.dig(:session, :password)
-      log_in @user
-      params.dig(:session, :remember_me) == "1" ? remember(@user) : forget(@user)
-      redirect_back_or @user
-    else
-      flash.now[:danger] = t("layouts.messages.invalid_email_password_combination")
-      render :new, status: :unprocessable_entity
-    end
+    reset_session
+    params.dig(:session, :remember_me) == "1" ? remember(@user) : forget(@user)
+    log_in @user
+    redirect_back_or @user
   end
 
   def destroy
     log_out
-    redirect_to root_url, status: :see_other
+    redirect_to root_url
   end
 
   private
 
   def load_user
-    @user = User.find_by(email: params.dig(:session, :email)&.downcase)
+    @user = User.find_by email: params.dig(:session, :email)&.downcase
     return if @user
 
     flash.now[:danger] = t("layouts.messages.user_not_found")
-    render :new, status: :unprocessable_entity
+    render :new
+  end
+
+  def check_authenticate
+    return if @user.authenticate(params.dig(:session, :password))
+
+    flash.now[:danger] = t("layouts.messages.invalid_email_password_combination")
+    render :new
+  end
+
+  def check_activation
+    return unless @user && !@user.activated?
```

Comment:
```suggestion
    return unless !@user.activated?
```
ri nhỉ, mà sao method check_activation lặp thế cu nhỉ